### PR TITLE
More configurable move distances

### PR DESCRIPTION
**Pull request by ellensp**
### Description

User with a very large build volume of 2000x2000x2000 has requested larger moves in move menu

Added VERY_LARGE_MOVE_ITEMS to Configuration_adv.h which is also automatically set when any axis is >=  2000
Enables menu    `SUBMENU(MSG_MOVE_500MM, []{ _goto_manual_move(500); });`

NOTE: to allow Z values greater than 999 I modified ftostr52sp to make the initial space optional.   

### Requirements

X_BED_SIZE >= 2000
and/or 
Y_BED_SIZE >= 2000
and/or
Z_MAX_POS >= 2000
or #define VERY_LARGE_MOVE_ITEMS

With a marlinui display, such as a 128x64 mono lcd

### Benefits

Adds additional large moves on very large build volumes

### Configurations

Ramps with REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER
and #define X_BED_SIZE 2000, #define Y_BED_SIZE 2000, #define Z_MAX_POS 2000

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/12908535/Configuration.zip)

### Related Issues

https://github.com/MarlinFirmware/Marlin/issues/26337

### CODE REVIEW

**Comment by mikemerryguy** [comment]
I personally added the 50 and 100mm and .5in 1in moves
there is a lot more you have to do here remembering back!
You have to add the move to EVERY language file!
It was not that hard, compare the strings in english to what you want in the other language, if the translation isn't correct it will be addressed by someone of that native language, but if you actually look at the lines, it mostly just copy and paste!!!

**Comment by ellensp** [comment]
"You have to add the move to EVERY language file!" no you do not, all languages default to using the english strings if not found.

If users want it in their language, they should contribute and create PRs to add them.

**Comment by ellensp** [comment]
LARGE_MOVE_ITEMS already adds in 50 and 100mm etc
This is VERY_LARGE_MOVE_ITEMS Ie bigger still

**Comment by thinkyhead** [comment]
I'm thinking refactor, what about you? Let's replace these settings with a single setting, with a list of move lengths, and if special handling is needed for very short ones (e.g., more immediate response) then we can work that out as part of the package. Something like this:

```cpp
// Manual move distances to show in the menus. Comma-separated list
#define MANUAL_MOVE_DISTANCE_MM 0.1, 1.0, 10.0, 50.0, 100.0, 500.0 // (mm)
#define MANUAL_MOVE_DISTANCE_IN 0.001, 0.01, 0.1, 0.5, 1.0, 5.0 // (in) For INCH_MODE_SUPPORT
```

**Comment by thinkyhead** [comment]
Okay! The use of a bare list is turning out to be good for the `MAP` macro. We'll have to keep that in mind for future needs.

Anyway, this turns out to make the configuration more straightforward, it also makes it totally expandable and simplifies the code. So as refactors go this looks like a win all 'round.

**Comment by ellensp** [comment]
"I'm thinking refactor"  you always do...

File: Marlin/src/libs/numtostr.cpp [hunk 1]
```diff
@@ -413,7 +413,7 @@ const char* ftostr72rj(const_float_t f) {
 const char* ftostr52sp(const_float_t f) {
   long i = INTFLOAT(f, 2);
   uint8_t dig;
-  conv[1] = MINUSOR(i, ' ');
+  conv[1] = RJDIGIT(i, 100000);
```
Comment (thinkyhead) [inline Marlin/src/libs/numtostr.cpp]: Whatever is printing in a non-ideal way, maybe it should use `ftostr52sprj` or another num-to-string function that better fits.

**Comment by ellensp** [comment]
This does not work in simulator

Config changes from stock.

#define MOTHERBOARD BOARD_SIMULATED
#define REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER

Yes Configuration_adv.h contains
#define MANUAL_MOVE_DISTANCE_MM                    10, 1.0, 0.1  // (mm)
 
This is what you get
missing

This is what was  expected
expected

**Comment by thisiskeithb** [comment]
> This does not work in simulator

Doesn't work with `TFT_COLOR_UI` either.

**Comment by ellensp** [comment]
Or 20x4

**Pull request merged**